app/FirebaseApi.js

```diff
@@ -0,0 +1,46 @@
+/* global someFunction window:true */
+
+import firebase from 'firebase';
+import RNFetchBlob from 'react-native-fetch-blob';
+import Config from 'react-native-config';
+
+// Init Firebase
+const config = {
+  apiKey: Config.FIREBASE_API_KEY,
+  authDomain: 'freemarketrn.firebaseapp.com',
+  storageBucket: 'freemarketrn.appspot.com',
+};
+firebase.initializeApp(config);
+const storage = firebase.storage();
+
+// Prepare Blob support
+const Blob = RNFetchBlob.polyfill.Blob;
+const fs = RNFetchBlob.fs;
+window.XMLHttpRequest = RNFetchBlob.polyfill.XMLHttpRequest;
+window.Blob = Blob;
+
+export const saveImageToStorage = localImage => (
+  new Promise((resolve, reject) => {
+    const uploadUri = localImage.uri;
+    const mime = 'application/octet-stream';
+    const sessionId = new Date().getTime();
+    let uploadBlob = null;
+    const imageRef = storage.ref('images').child(`${sessionId}${localImage.fileName}`);
+    fs.readFile(uploadUri, 'base64')
+      .then(data => Blob.build(data, { type: `${mime};BASE64` }))
```

Comment:
First of all you should have already read data in `localImage.data`, in Base64.
So you don't need `RNFetchBlob`, nor `Blob`, nor `window` and I don't understand why even xhr is here.
Seems like Firebase ref's object `put` can take ByteArray: https://firebase.google.com/docs/storage/web/upload-files#upload_from_a_byte_array,
try to use strict `localImage.data` and if it'll fail try `Uint8Array(localImage.data)`